### ISSUE TITLE
`boolean-positional-value-in-call` (`FBT003`) - don't warn when function name starts with `set_` and only takes one argument

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8923]
functions like these should not report the error imo, since the meaning of the bool is self explanatory, which is often the case when the functin starts with `set_`:

```py
def set_enabled(enabled: bool):
    ...

set_enabled(True) # self explanatory
set_enabled(enabled=True) # unnecessary/redundant
```

**Comment by charliermarsh** [comment]
We may want to do this, it would also help with https://github.com/astral-sh/ruff/issues/9287 and https://github.com/astral-sh/ruff/issues/3247#issuecomment-1842529367.

**Comment by tmke8** [comment]
Maybe also allow the `use_` prefix? I encountered that with [torch.use_deterministic_algorithms](https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html). The following feels a bit silly:

```python
torch.use_deterministic_algorithms(mode=True)
```

**Comment by bswck** [comment]
I often use flag-holding context vars.
```py
from contextvars import ContextVar

cv = ContextVar("cv", default=False)
cv.set(True)  # triggers FBT003
```

What a pity the fix for this doesn't apply to `ContextVar.set`.

**Comment by DetachHead** [comment]
it looks like the check expects the function name to start with "set" followed by either an underscore or an uppercase character, but not "set" on its own. it should probably be updated to allow that too.

**Comment by zanieb** [comment]
It seems reasonable to expand it to cover that case.
